pkgs/mkService.nix

```diff
@@ -113,7 +115,9 @@ let
     if user.name == "root"
       then exec
       else
-        let suf = lib.optionalString (user.groups != []) ":${lib.concatStringsSep ":" user.groups}";
+        let
+          allGroups = (builtins.concatLists [ user.groups user.userGroups ]);
+          suf = lib.optionalString (allGroups != []) ":${lib.concatStringsSep ":" allGroups}";
```

Comment:
`lib.optionalString (l != []) ":${lib.concatStringsSep ":" l}"` ≡ `lib.concatMapStrings (i: ":${i}") l`

(This one is my fault.)